tests/federation_room_send_test.go

```diff
@@ -249,3 +249,51 @@ func TestNetworkPartitionOrdering(t *testing.T) {
 		},
 	})
 }
+
+// Tests that /sync v2 returns the correct timeline for the following situation _over federation_:
+//   - Alice and Bob in a room.
+//   - Bob leaves the room.
+//   - Alice sends a message.
+//   - Bob joins the room.
+//   - The next sync response should include a timeline which ends with [BOB_LEAVE, ALICE_MSG, BOB_JOIN].
```

Comment:
I think it would be perfectly valid to return just `BOB_JOIN` here, which I thought we did